### DISCUSSION TITLE
Where can I get the django stubs bundled with pylance?

### DISCUSSION THREAD

**Discussion by lexicalunit** [microsoft/pylance-release#2832] [Q&A] [answered]
I'd like to be able to run `pyright` on my command line and get the same results I do in my editor. I tried installing https://github.com/sbdchd/django-types but that didn't do it, I also tried https://github.com/typeddjango/django-stubs but no luck there either.

Exactly what the heck is the bundled stubs and how do they seem to work so well? Where can I get them?

**Comment by erictraut** [comment, marked as answer]
The stubs that ship with pylance are located [here](https://github.com/microsoft/python-type-stubs).

**Reply by lexicalunit** [reply]
Thank you. I was able to import these stubs into our project using a git submodule.

```shell
cd our-project
git clone --depth=1 --no-checkout https://github.com/microsoft/python-type-stubs stubs
git submodule add https://github.com/microsoft/python-type-stubs stubs
git submodule absorbgitdirs
git -C stubs config core.sparseCheckout true
echo 'django' >>.git/modules/stubs/info/sparse-checkout
git submodule update --force --checkout stubs
```

We can subsequently update to the latest stubs by running:
```shell
git submodule update --force --checkout stubs
```

If you know off the top of your head how I can get around this and instead do something like `pipenv install --dev python-type-stubs[django]` (just a made up incantation for demonstration purposes), please let me know.
